lib/kitchen/loader/yaml.rb

```diff
@@ -192,6 +192,10 @@ def read_file(file)
       # @return [String] an absolute path to a Kitchen config YAML file
       # @api private
       def default_config_file
+        if File.exist?(kitchen_yml) && File.exist?(dot_kitchen_yml)
+          raise UserError, "Both #{kitchen_yml} and #{dot_kitchen_yml} found. Use one or the other. Test-Kitchen prefers the not-hidden filename."
+        end
```

Comment:
"Please use one or the other. The hidden file will be deprecated in the future."